test/classes/phing/tasks/ext/ComposerTaskTest.php

```diff
@@ -72,18 +72,63 @@ public function testSetGetPhp()
 
     /**
      * @covers ComposerTask::setComposer
+     */
+    public function testSetComposer()
+    {
+        $composer = 'foobar';
+        $o = $this->object;
+        $o->setComposer($composer);
+
+        $prop = new ReflectionProperty('ComposerTask', 'composer');
+        $prop->setAccessible(true);
+
+        $this->assertEquals($composer, $prop->getValue($o));
+    }
+
+    /**
      * @covers ComposerTask::getComposer
      */
-    public function testSetGetComposer()
+    public function testGetComposerNotOnPath()
+    {
+        $composer = 'bar';
+        $o = $this->object;
+
+        $orgPath = getenv("PATH");
+
+        $prop = new ReflectionProperty('ComposerTask', 'composer');
+        $prop->setAccessible(true);
+        $prop->setValue($o, $composer);
+
+        putenv("PATH=/foo/bar");
+
+        $pathComposer = $o->getComposer();
+
+        putenv("PATH=$orgPath");
+
+        $this->assertEquals($composer, $pathComposer);
+    }
+
+    /**
+     * @covers ComposerTask::getComposer
+     */
+    public function testGetComposerFromPath()
     {
         $composer = 'foo';
         $o = $this->object;
         $o->setComposer($composer);
-        $composerFile = new SplFileInfo($composer);
-        if (false === $composerFile->isFile()) {
-            $composer = FileSystem::getFileSystem()->which('composer');
-        }
-        $this->assertEquals($composer, $o->getComposer());
+
+        $testPath = PHING_TEST_BASE . '/etc/tasks/ext/composer';
+        $orgPath = getenv("PATH");
+
+        $pathSeparator = FileSystem::getFileSystem()->getPathSeparator();
+        putenv("PATH=$testPath$pathSeparator$orgPath");
+
+        $pathComposer = $o->getComposer();
+
+        putenv("PATH=$orgPath");
```

Comment:
AFAIK windows uses `Path` and not `PATH` 🤔 ?

Reply:
Windows isn't case sensitive so Path, PATH, and pAtH would all work.